tt_metal/hw/inc/dataflow_api.h

```diff
@@ -1514,8 +1591,16 @@ void noc_async_read_barrier(uint8_t noc = noc_index) {
 FORCE_INLINE
 void noc_async_write_barrier(uint8_t noc = noc_index) {
     WAYPOINT("NWBW");
-    while (!ncrisc_noc_nonposted_writes_flushed(noc))
-        ;
+    uint32_t i = 1;
+    while (!ncrisc_noc_nonposted_writes_flushed(noc)) {
+        if (i == 6400000) {
```

Comment:
debug code can be removed

Reply:
Still WIP, will remove before declaring it done